docs/propagation.md

```diff
@@ -210,3 +217,39 @@ along F0, propagate `["c"]` along F1, and propagate nothing along F2.
 T0  | "a", **"b"** | **"c"**  | "f" |
 T1  | "a", "b"     | "c", "d" | "g" |
 T2  | **"a", "b"** | "c", "e" |     |
+
+### Data flow edge ops
+
+The above propagation step description applies to most ops. However, there are
+cases where a sharding rule is not appropriate. For those cases, Shardy defines
+_data flow edge_ ops.
+
+A data flow edge of some op X defines a bridge between a set of *sources* and a
+set of *targets*, such that all sources and targets should be sharded in the
+same way. Examples of such ops are `OptimizationBarrierOp`, `WhileOp`, and
+`CaseOp`. Ultimately, any op that implements [ShardableDataFlowOpInterface](sdy_op_interfaces#shardabledataflowopinterface_shardabledataflowopinterface)
```

Comment:
Those ops are from StableHLO, shall we mention that?

Reply:
+1 would say `stablehlo::OptimizationBarrierOp`, `stablehlo::WhileOp`, `stablehlo::CaseOp`, but also `sdy::ManualComputationOp` to show our internal ops are using it too (`NamedComputationOp` internally does as well, but we don't really advertise that op as much as `ManualComputationOp`).